pkg/cmd/label/list.go

```diff
@@ -36,17 +43,33 @@ func newCmdList(f *cmdutil.Factory, runF func(*listOptions) error) *cobra.Comman
 	}
 
 	cmd := &cobra.Command{
-		Use:     "list",
-		Short:   "List labels in a repository",
-		Long:    "Display labels in a GitHub repository.",
+		Use:   "list",
+		Short: "List labels in a repository",
+		Long: heredoc.Docf(`
+		Display labels in a GitHub repository.
+
+		You cannot use %[1]s--search%[1]s with either %[1]s--order%[1]s or %[1]s--sort%[1]s.
```

Comment:
```suggestion
			When using the %[1]s--search%[1]s flag results are sorted by best match of the query.
```
We don't usually put flag restrictions in the documentation and rely on the error returned to inform the user of the restriction. What about this suggestion that defines the behavior when `--search` is used instead?

Reply:
Some commands do list restrictions in the long description. I found several when seeing if it would be consistent to put constraints in the flag documentation (what I actually see most CLIs do when the constraint is simple) or in the long description. Just searching for "cannot use" (hence my original wording),

* pkg/cmd/codespaces/ssh.go
* pkg/cmd/extension/command.go
* pkg/cmd/pr/checks/checks.go

Is there a good reason to not inform the customer ahead of time of known client-side constraints? Seems poorly documented otherwise, leaving them to discover constraints - much like my command above about empty strings being okay in some places vs. not in others for the GraphQL API. "Just run it and find out" against production data doesn't seem like a great UX.

Reply:
Quickly looking at the three you listed it seems that they are all errors returned when the commands are executed not in the descriptions of the commands. 

I know better flag documentation, regarding this exact scenario, is something the team has talked about in the past, but we haven't come up with a good strategy yet. The counter argument for listing the flag constraints in the description is that it is not scalable and is prone to human error, either forgetting to add it or forgetting to update it if the actual constraint changes.

Reply:
I was just about to make the change but I see you already did. 🙏